### PR TITLE
Code changes to handle form-data

### PR DESCRIPTION
**Pull request by sarangan12**
This is to address the issue https://github.com/Azure/autorest.typescript/issues/679. 

This handles the body-formdata.json. I will discuss with @joheredi regarding the urlencoded version. 

@joheredi Please review and approve

@ramya-rao-a FYI....

### CODE REVIEW

File: src/transforms/operationTransforms.ts [hunk 1]
```diff
@@ -478,23 +480,43 @@ function getGroupedParameters(
...
-        p.location === ParameterLocation.Uri
-    ),
-    headerParameters: operationParams.filter(
-      p => p.location === ParameterLocation.Header
-    ),
-    cookie: operationParams.filter(p => p.location === ParameterLocation.Cookie)
-  };
+  if (mediaType && (mediaType == KnownMediaType.Multipart || mediaType == KnownMediaType.Form)) {
+    return {
+      formDataParameters: operationParams.filter(
```
Comment (joheredi) [inline src/transforms/operationTransforms.ts:485]: you could also do this without the if/else with something like

```typescript
const hasFormDataParameters = mediaType && (mediaType == KnownMediaType.Multipart || mediaType == KnownMediaType.Form);

return {
   ...(hasFormDataParameters  
         ? {formDataParameters: operationParams.filter(p => p.location === ParameterLocation.Body)}
         : {requestBody: operationParams.find(p => p.location === ParameterLocation.Body)} ),
      queryParameters: operationParams.filter(
        p => p.location === ParameterLocation.Query
      ),
      urlParameters: operationParams.filter(
        p =>
          p.location === ParameterLocation.Path ||
          p.location === ParameterLocation.Uri
      ),
      headerParameters: operationParams.filter(
        p => p.location === ParameterLocation.Header
      ),
      cookie: operationParams.filter(p => p.location === ParameterLocation.Cookie)
}
```

File: test/integration/bodyFormData.spec.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        if (result.readableStreamBody) {
+            const buff = await readStreamToBuffer(result.readableStreamBody)
+            assert.deepEqual(buff, fileContent);
+        } else {
+            assert.fail("ReadableStreamBody must not be null!!!");
+        }
+    });
+});
+
+
```
Comment (joheredi) [inline test/integration/bodyFormData.spec.ts:41]: Would it be possible to add one more test similar to this one? https://github.com/Azure/autorest.typescript/blob/922fbd9d63fff86dbb0b59814118178400b3a493/test/vanilla/acceptanceTests.ts#L1954

**Review by joheredi — APPROVED** [review]
Looks great! Thanks for taking this work. If possible please add one more test similar to this one before merging  https://github.com/Azure/autorest.typescript/blob/922fbd9d63fff86dbb0b59814118178400b3a493/test/vanilla/acceptanceTests.ts#L1954

File: src/transforms/operationTransforms.ts (hunk 1 above)
Comment (sarangan12) [reply src/transforms/operationTransforms.ts]: Done

File: test/integration/bodyFormData.spec.ts (hunk 2 above)
Comment (sarangan12) [reply test/integration/bodyFormData.spec.ts:73]: Done

**Pull request merged**